File: PatientRecords.BLLayer/EntityViews/PatientRecordView.cs


using PatientRecords.BLLayer.EntityDTOs;
using System;

namespace PatientRecords.BLLayer.EntityViews
{
    public class PatientRecordView 
    {
        public int Id { get; set; }
        public string DiseaseName { get; set; }
        public DateTime TimeOfEntry { get; set; }
        public string Description { get; set; }
        public decimal Bill { get; set; }
        public int PatientId { get; set; }
        public PatientView Patient { get; set; }
        public DateTime CreatedDate { get; set; }
        public DateTime UpdatedDate { get; set; }
        public string UpdatedBy { get; set; }
        public string CreatedBy { get; set; }
        public string SearchField { get; set; }

    }
}

File: PatientRecords.ReactJS/src/redux/patientRecords/models/patientRecordView.js
import { BaseModel } from 'sjs-base-model';
import pationtView from '../../pationts/models/pationtView';

export default class PatientRecordView extends BaseModel {
    id = 0;
    patientId = 0;
    diseaseName = '';
    bill = 0.0;
    description = '';
    patientDTO = pationtView;
    timeOfEntry =  Date;
    createdDate = new Date;
    updatedDate = new Date;
    updatedBy = '';
    createdBy = '';
    constructor(data) {
      super();
  
      this.update(data);
    }
  }